README.md

```diff
@@ -1,57 +1,142 @@
-# Project Name
+# OpenAI end-to-end basic reference implementation
 
-(short, 1-3 sentenced, description of the project)
+This reference implementation illustrates a basic approach for authoring and running a chat application in a single region with Azure Machine Learning and OpenAI.
 
-## Features
+The implementation takes advantage of [Prompt flow](https://microsoft.github.io/promptflow/) in [Azure Machine Learning](https://azure.microsoft.com/products/machine-learning) to build and deploy flows that can link the following actions required by an LLM chat application:
 
-This project framework provides the following features:
+- Creating prompts
+- Querying data stores for grounding data
+- Python code
+- Calling Large Language Models (LLMs)
 
-* Feature 1
-* Feature 2
-* ...
+The reference implementation illustrates a basic example of a chat application.  For a reference implementation that implements enterprise requirements, please see the [OpenAI end-to-end baseline reference implementation](https://github.com/Azure-Samples/openai-end-to-end-baseline).
 
-## Getting Started
+## Architecture
+
+The implementation covers the following scenarios:
+
+1. Authoring a flow - Authoring a flow using Prompt flow in an Azure Machine Learning workspace.
+1. Deploying a flow - The client UI is hosted in Azure App Service and accesses the OpenAI service via a Machine Learning managed online endpoint.
+
+### Deploying a flow to Azure Machine Learning managed online endpoint
+
+![Diagram of the deploying a flow to Azure Machine Learning managed online endpoint.](docs/media/openai-end-to-end-basic.png)
+
+The Azure Machine Learning deployment architecture diagram illustrates how a front-end web application connects to a managed online endpoint.
+
+## Deploy
+
+The following are prerequisites.
 
 ### Prerequisites
 
-(ideally very short, if any)
+- Ensure you have an [Azure Account](https://azure.microsoft.com/free/)
+- The deployment must be started by a user who has sufficient permissions to assign [roles](https://learn.microsoft.com/azure/role-based-access-control/built-in-roles), such as a User Access Administrator or Owner.
+- Ensure you have the [Azure CLI installed](https://learn.microsoft.com/cli/azure/install-azure-cli)
+- Ensure you have the [az Bicep tools installed](https://learn.microsoft.com/azure/azure-resource-manager/bicep/install)
+- Ensure the account working with the Azure Machine Learning Workspace has been granted the 'Cognitive Services OpenAI Contributor' role
```

Comment:
> - Ensure the account working with the Azure Machine Learning Workspace has been granted the 'Cognitive Services OpenAI Contributor' role

Where is this needed?  If it is, can you please make this a step in the instructions?  Otherwise, as a pre-req, where would you place this?  On the subscription?

Reply:
Updated (not subscription, OpenAI instance)